### PR TITLE
[d3d9, dxso] Alias color and depth samplers and improve tracking 

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Takes me from 340 -> 460fps in A Hat in Time's main menu when CPU bound.

Leaving as draft until it's been tested, large and dangerous change.

### CODE REVIEW

**Comment by Askannz** [comment]
Rayman Origins crashes on startup.

Proton log:
[steam-207490.log](https://github.com/doitsujin/dxvk/files/6950548/steam-207490.log)

Proton: 6.3-5
GPU: RX 6800
Driver: Radv, Mesa 21.1.6
Distribution: Arch
Kernel: 5.11.7-139-tkg-pds

**Comment by Joshua-Ashton** [comment]
Please re-test, should be fixed now.

**Comment by Askannz** [comment]
Woks now! I've only played through a short section but everything seems fine.

**Comment by lukashoracek** [comment]
Metro 2033 (non-redux) works now after the fix.

File: src/d3d9/d3d9_texture.h [hunk 1]
```diff
@@ -196,17 +196,17 @@ namespace dxvk {
...
-      return static_cast<D3D9Texture2D*>  (ptr)->GetCommonTexture();
-    else if (type == D3DRTYPE_CUBETEXTURE)
-      return static_cast<D3D9TextureCube*>(ptr)->GetCommonTexture();
-    else //if(type == D3DRTYPE_VOLUMETEXTURE)
-      return static_cast<D3D9Texture3D*>  (ptr)->GetCommonTexture();
+    // We can avoid needing to get the type as m_texture has the same offset
+    // no matter the texture type.
+    // The compiler is not smart enough to eliminate the call to GetType as it is
+    // not marked const.
+    return static_cast<D3D9Texture2D*>(ptr)->GetCommonTexture();
```
[the remark this answers is not in the document]
Comment (Joshua-Ashton) [reply src/d3d9/d3d9_texture.h:209]: Me too but GetType showed up in perf top (fairly low down but still...) for some games and this is why.

**Pull request merged**